src/bam_io.cpp

```diff
@@ -88,6 +118,31 @@ bool BamCramReader::SetRegion(const std::string& chrom, int32_t start, int32_t e
   }
 }
 
+void BamAlignment::TrimEnd(int32_t trim_to) {
+  // method for trimming reads to a defined length
+  // use-case example: mixed readlength sequencing data
+
+  // update length if read length is larger than trim_to; otherwise do nothing
+  if (b_->core.l_qseq > trim_to && trim_to >= 1) {
+    length_ = trim_to;
```

Comment:
should we put an else clause here? To jump out? Or do something appropriate?

Reply:
commenting again because apparently I wasn't commenting in review mode before lol.

Reply:
If the read is not longer then trim_to, then no fields need to be updated :) The method is void so basically there would be no effect on the BamAlignment object if its length is larger than trim_to or if the user passed a negative value or a zero as trim_to, because those cases don't make sense. trim_to is -1 by default even if user doesn't pass that option.

What do youthink?

Reply:
I agree that's the correct usage of this function, but I'm thinking if an incorrect usage may cause an error. I think it's worth adding an else to avoid weird corner cases that we may have not thought about.

Reply:
I see what you mean. Good point. I added some clauses below to catch other contingencies. I think that covers all possible ranges of trim_to. I tried to arrange the clauses in order of more frequent to least frequent so that we don't take a performance hit.

Do you think there are any other ones worth adding?